src/macaron/slsa_analyzer/checks/build_as_code_check.py

```diff
@@ -130,164 +86,110 @@ def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResu
-                            if ctx.dynamic_data["is_inferred_prov"] and ci_info["provenances"]:
-                                predicate = ci_info["provenances"][0]["predicate"]
-                                predicate["buildType"] = f"Custom {ci_service.name}"
-                                predicate["builder"]["id"] = config_name
-                                predicate["invocation"]["configSource"][
-                                    "uri"
-                                ] = f"{ctx.remote_path}@refs/heads/{ctx.branch_name}"
-                                predicate["invocation"]["configSource"]["digest"]["sha1"] = ctx.commit_sha
-                                predicate["invocation"]["configSource"]["entryPoint"] = config_name
-                            check_result["result_tables"] = [
-                                BuildAsCodeTable(
-                                    build_tool_name=build_tool.name,
-                                    ci_service_name=ci_service.name,
-                                    deploy_command=deploy_kw,
-                                )
-                            ]
-                            return CheckResultType.PASSED
+                # Populate the BuildAsCodeSubchecks object with the certainty results from subchecks.
+                build_as_code_subchecks.build_as_code_subcheck_results = BuildAsCodeSubchecks(ctx=ctx, ci_info=ci_info)
+
+                prolog_string = PrologString(
+                    """
+                    :- use_module('src/macaron/slsa_analyzer/checks/problog_predicates.py').
+
+                    A :: ci_parsed :- ci_parsed_check(A).
+                    B :: deploy_action :- deploy_action_check(B).
+                    C :: deploy_command :- deploy_command_check(C).
+                    D :: deploy_kws :- deploy_kws_check(D).
+
+                    0.80 :: deploy_action_certainty :- deploy_action.
+                    0.15 :: deploy_action_certainty :- deploy_action, ci_parsed.
+
+                    0.70 :: deploy_command_certainty :- deploy_command.
+                    0.15 :: deploy_command_certainty :- deploy_command, ci_parsed.
+
+                    0.60 :: deploy_kws_certainty :- deploy_kws.
+
+                    build_as_code_check :- deploy_action_certainty; deploy_command_certainty; deploy_kws_certainty.
+
+                    query(build_as_code_check).
+                    """
+                )
+
+                # TODO: query each of the methods, and take the values from the one with the highest confidence.
+                confidence_score = 0.0
+                result = get_evaluatable().create_from(prolog_string).evaluate()
+                for key, value in result.items():
+                    print(key, value)
+                    if str(key) == "build_as_code_check":
+                        confidence_score = float(value)
+                results = vars(build_as_code_subchecks.build_as_code_subcheck_results)
```

Comment:
It's okay to use this built in function for now. But we should keep in mind that this function would raise TypeError when `__dict__` attribute could not be found within the target object.

Reply:
Thanks for your comment Nhan, I'm actually about to push a change that will mean I'm no longer using this anyway so hopefully it's all good.